// ngx-analytics/bundles/kissmetrics.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('ngx-analytics')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', 'ngx-analytics'], factory) :
	(factory((global['ngx-analytics'] = global['ngx-analytics'] || {}, global['ngx-analytics'].kissmetrics = {}),global.ng.core,global['ngx-analytics']));
}(this, (function (exports,core,ngxAnalytics) { 'use strict';

var NgxAnalyticsKissmetrics = (function () {
    function NgxAnalyticsKissmetrics(ngxAnalytics$$1) {
        var _this = this;
        this.ngxAnalytics = ngxAnalytics$$1;
        if (typeof (_kmq) === 'undefined') {
            _kmq = [];
        }
        this.ngxAnalytics.pageTrack.subscribe(function (x) { return _this.pageTrack(x.path); });
        this.ngxAnalytics.eventTrack.subscribe(function (x) { return _this.eventTrack(x.action, x.properties); });
        this.ngxAnalytics.setUsername.subscribe(function (x) { return _this.setUsername(x); });
        this.ngxAnalytics.setUserProperties.subscribe(function (x) { return _this.setUserProperties(x); });
    }
    NgxAnalyticsKissmetrics.prototype.pageTrack = function (path) {
        _kmq.push(['record', 'Pageview', { 'Page': path }]);
    };
    NgxAnalyticsKissmetrics.prototype.eventTrack = function (action, properties) {
        _kmq.push(['record', action, properties]);
    };
    NgxAnalyticsKissmetrics.prototype.setUsername = function (userId) {
        _kmq.push(['identify', userId]);
    };
    NgxAnalyticsKissmetrics.prototype.setUserProperties = function (properties) {
        _kmq.push(['set', properties]);
    };
    NgxAnalyticsKissmetrics.decorators = [
        { type: core.Injectable },
    ];
    NgxAnalyticsKissmetrics.ctorParameters = function () { return [
        { type: ngxAnalytics.NgxAnalytics, },
    ]; };
    return NgxAnalyticsKissmetrics;
}());

exports.NgxAnalyticsKissmetrics = NgxAnalyticsKissmetrics;

Object.defineProperty(exports, '__esModule', { value: true });

})));
