// ng2-adsense/esm5/adsense.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ADSENSE_TOKEN } from './adsense-config';
import { AdsenseComponent } from './adsense.component';
var AdsenseModule = /** @class */ (function () {
    function AdsenseModule() {
    }
    /**
     * @param {?=} config
     * @return {?}
     */
    AdsenseModule.forRoot = /**
     * @param {?=} config
     * @return {?}
     */
    function (config) {
        if (config === void 0) { config = {}; }
        return {
            ngModule: AdsenseModule,
            providers: [{ provide: ADSENSE_TOKEN, useValue: config }],
        };
    };
    AdsenseModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule],
                    exports: [AdsenseComponent],
                    declarations: [AdsenseComponent],
                },] }
    ];
    return AdsenseModule;
}());
export { AdsenseModule };
